src/test/bufferlist.cc

```diff
@@ -282,6 +282,55 @@ class TestRawPipe : public ::testing::Test {
 };
 int TestRawPipe::fd = 0;
 
+ssize_t safe_copy_to_fd(const bufferptr& ptr, int fd, off_t ofs_from, ssize_t req_size, off_t dst_offset = -1) {
+  ssize_t r;
+  ssize_t count = 0;
+  while (req_size > 0) {
+    r = ptr.copy_to_fd(fd, ofs_from + count, req_size, dst_offset);
+    if (r == -EAGAIN) continue;
```

Comment:
nit, instead of busy waiting, might want to wait using `select()` or `poll()` before trying again.